### ISSUE TITLE
dplyr 0.8.0 RC breaks sample_n.tbl_dt

### ISSUE THREAD

**Issue by romainfrancois** [tidyverse/dtplyr#63]
<details>
```
> revdep_details(revdep = "dtplyr")
══ Reverse dependency check ═══════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════ dtplyr 0.0.2 ══

Status: BROKEN

── Newly failing

✖ checking tests ...

── Before ─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
0 errors ✔ | 0 warnings ✔ | 0 notes ✔

── After ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
❯ checking tests ...
  See below...

── Test failures ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────── testthat ────

> library(testthat)
> library(dtplyr)
> 
> test_check("dtplyr")
── 1. Error: sample preserves class (@test-sample.R#4)  ────────────────────────
argument "e2" is missing, with no default
1: expect_is(sample_n(data.table(mtcars), 1), "data.table") at testthat/test-sample.R:4
2: quasi_label(enquo(object), label)
3: eval_bare(get_expr(quo), get_env(quo))
4: sample_n(data.table(mtcars), 1)
5: sample_n.data.frame(data.table(mtcars), 1) at /Users/romain/git/tidyverse/dplyr/R/sample.R:53
6: slice(tbl, sample.int(n(), check_size(!!size, n(), replace = replace), replace = replace, 
       prob = !!weight)) at /Users/romain/git/tidyverse/dplyr/R/dataframe.R:268
7: slice.data.table(tbl, sample.int(n(), check_size(!!size, n(), replace = replace), 
       replace = replace, prob = !!weight)) at /Users/romain/git/tidyverse/dplyr/R/manip.r:169
8: slice_(.data, .dots = lazyeval::lazy_dots(...)) at /Users/romain/git/release/dplyr/revdep/checks.noindex/dtplyr/new/dtplyr.Rcheck/00_pkg_src/dtplyr/R/tbl-dt.r:314
9: slice_.data.table(.data, .dots = lazyeval::lazy_dots(...)) at /Users/romain/git/tidyverse/dplyr/R/manip.r:178
10: dt_subset(.data, , j, env) at /Users/romain/git/release/dplyr/revdep/checks.noindex/dtplyr/new/dtplyr.Rcheck/00_pkg_src/dtplyr/R/tbl-dt.r:329
11: eval(call, env) at /Users/romain/git/release/dplyr/revdep/checks.noindex/dtplyr/new/dtplyr.Rcheck/00_pkg_src/dtplyr/R/utils.R:22
12: eval(call, env)
13: `_dt`[, .SD[sample.int(.N, check_size(!!size, .N, replace = replace), replace = replace, 
       prob = !!weight)], by = `_vars`]
14: `[.data.table`(`_dt`, , .SD[sample.int(.N, check_size(!!size, .N, replace = replace), 
       replace = replace, prob = !!weight)], by = `_vars`)
15: `[.data.frame`(x, i, j)
16: sample.int(.N, check_size(!!size, .N, replace = replace), replace = replace, prob = !!weight)
17: Ops.quosure(weight)
18: is_quosure(e2)

══ testthat results  ═══════════════════════════════════════════════════════════
OK: 127 SKIPPED: 0 FAILED: 1
1. Error: sample preserves class (@test-sample.R#4) 

Error: testthat unit tests failed
Execution halted
```
</details>

**Comment by hadley** [comment]
Minimal reprex:

``` r
library(dtplyr)
library(dplyr, warn.conflicts = FALSE)

sample_n(data.table::data.table(x = 1:10), 1)
#> Error in `[.data.frame`(x, i, j): object '.SD' not found

dt <- tbl_dt(data.frame(x = 1:10))
sample_n(dt, 1)
#> Error in `[.data.frame`(x, i, j): object '.SD' not found
```

Although I see a different error when running interactively.  

`sample_n()` and `sample_frac()` need to be rewritten to use tidy eval.

**Comment by hadley** [comment]
Fixed in 801d8960cad6c247e34fbba5d2109ebdad8e0ead
